src/main/java/edu/cornell/kfs/sys/aws/batch/AwsSecretValidationStep.java

```diff
@@ -0,0 +1,167 @@
+        }
+
+        if (StringUtils.isNotBlank(errorMessage)) {
+            throw new RuntimeException(errorMessage);
+        } else {
+            LOG.info("confirmAmazonSecretValidationShared, the shared secrets contained the expected values");
+        }
+    }
+
+    private Map<String, Object> buildNewSecretValue() {
+        Map<String, Object> values = new HashMap<>();
+        values.put(ACCESS_TOKEN, UUID.randomUUID().toString());
+        values.put(REFRESH_TOKEN, UUID.randomUUID().toString());
+        values.put(ACCESS_TOKEN_EXPIRATION_DATE, Calendar.getInstance(Locale.US).getTime());
+        return values;
+    }
+    
+    protected void updateAmazonSecretValidationInstance(AmazonSecretValidationInstance instance,
+            Map<String, Object> newSecretValues) {
+        instance.setAccess_token(newSecretValues.get(ACCESS_TOKEN).toString());
+        instance.setRefresh_token(newSecretValues.get(REFRESH_TOKEN).toString());
+        instance.setAccess_token_expiration_date((Date) newSecretValues.get(ACCESS_TOKEN_EXPIRATION_DATE));
+    }
+
+    protected void confirmInstanceSecretValues(Map<String, Object> expectedSecretValues,
+            AmazonSecretValidationInstance instanceValues) {
+        String errorMessage = StringUtils.EMPTY;
+        if (!StringUtils.equals(instanceValues.getAccess_token(),
+                expectedSecretValues.get(ACCESS_TOKEN).toString())) {
+            errorMessage = errorMessage + "Did not update ACCESS token as expected.";
+        }
+        if (!StringUtils.equals(instanceValues.getRefresh_token(),
+                expectedSecretValues.get(REFRESH_TOKEN).toString())) {
+            errorMessage = errorMessage + " Did not update REFRESH token as expected.";
+        }
+        if (!instanceValues.getAccess_token_expiration_date()
+                .equals((Date) expectedSecretValues.get(ACCESS_TOKEN_EXPIRATION_DATE))) {
+            errorMessage = errorMessage + " Did not update access token EXPIRATION DATE as expected.";
+        }
+
+        if (StringUtils.isNotBlank(errorMessage)) {
+            throw new RuntimeException(errorMessage);
+        }
+    }
+    
+    protected void validateUsingNewAWSServiceFromSpring() {
+        AwsSecretService service = SpringContext.getBean(AwsSecretService.class);
+        try {
+            AmazonSecretValidationShared sharedSpaceSecrets = service.getPojoFromAwsSecret(AWS_SECRET_NAME_VALIDATION_SHARED, 
+                    false, AmazonSecretValidationShared.class);
+            confirmAmazonSecretValidationShared(sharedSpaceSecrets);
+            service.logCacheStatus();
+        } catch (JsonProcessingException e) {
+            LOG.error("validateUsingNewAWSServiceFromSpring, had an error calling AWS Secret Service ", e);
+            throw new RuntimeException(e);
+        }
+    }
+    
+    protected void waitFiveSecondsAfterUpdatingSecretAndClearingCacheBeforeRetrieveSecretFromAws() {
```

Comment:
How about "waitForAwsSecretUpdateToFullyPost" or something along those lines?

Reply:
that works

Reply:
@dp65 one of the goals is for the code to "speak for itself" as in comments are needed to clarify  the code ..... the function names will tell the next developer why we are adding a wait